src/main/java/digital/slovensko/autogram/core/visualization/DocumentVisualizationBuilder.java

```diff
@@ -77,6 +94,79 @@ private Visualization createVisualization(SigningJob job, MimeType transformatio
         return new UnsupportedVisualization(job);
     }
 
+    private DSSDocument getOriginalDocument(DSSDocument document) {
+        SignedDocumentValidator documentValidator = SignedDocumentValidator.fromDocument(document);
+        documentValidator.setCertificateVerifier(new CommonCertificateVerifier());
+        List<AdvancedSignature> signatures = documentValidator.getSignatures();
+        if (signatures.isEmpty()) {
+            throw new RuntimeException("No signatures in document");
+        }
+        AdvancedSignature advancedSignature = signatures.get(0);
+        List<DSSDocument> originalDocuments = documentValidator.getOriginalDocuments(advancedSignature.getId());
+        if (originalDocuments.isEmpty()) {
+            throw new RuntimeException("No original documents found");
+        }
+        return originalDocuments.get(0);
+    }
```

Comment:
Toto sa nedá spraviť bez podpisov? Lebo teoreticky vie existovať aj .asice, v ktorom tá štruktúra, ale zatiaľ žiadny podpis. [DSS demo](https://ec.europa.eu/digital-building-blocks/DSS/webapp-demo/validation) to vie spracovať a povie, že je tam 0 podpis a taký a taký dokument. Trochu som pozeral dokumentáciu a tiež som nenašiel, že by sa to dalo cez tento DocumentValidator Interface. Nevieme spraviť výhybku, že ak 0 podpisov, tak nejakým asiceExtractorom to vybrať?

Reply:
Mohli by sme getnut vsetky subory z toho extractoru / asicContentu a potom hladat podla nazvu. Mozme to tym padom pouzit vzdy nemusi tam byt ze bud podla podpisov alebo z extractoru nie?

Reply:
@Bombino1024 ak vieš toto spraviť, tak ešte spravme. Nepodpísaný asic vieš spraviť tak, že z podpísaného vymažeš podpis (manifest/signatureXX.xml)